tracer.go

```diff
@@ -206,7 +206,10 @@ func (t *Tracer) startSpanWithOptions(
 		if hasParent && parent.isDebugIDContainerOnly() {
 			ctx.flags |= (flagSampled | flagDebug)
 			samplerTags = []Tag{{key: JaegerDebugHeader, value: parent.debugID}}
-			//samplerTags = []Tag{{key: t.headerKeys.JaegerDebugHeader, value: parent.debugID}}
+			textMapPropagator, ok := t.injectors[opentracing.TextMap].(*textMapPropagator)
+			if ok {
+				samplerTags = []Tag{{key: textMapPropagator.headerKeys.JaegerDebugHeader, value: parent.debugID}}
+			}
```

Comment:
this feel dirty but it is a workaround for this edge case. Note that this is a side affect of not having the tracer know about the header key values. This seems to be the only edge case in which the tracer would need to know the value of any of the headers.